lte/gateway/c/oai/tasks/nas/emm/Attach.c

```diff
@@ -2460,9 +2460,9 @@ void proc_new_attach_req(struct ue_mm_context_s* ue_context_p) {
        Do not send UE Context Release Command to eNB before receiving SGs IMSI
        Detach Ack from MSC/VLR */
     if (ue_context_p->sgs_context != NULL) {
-      if (((ue_context_p->sgs_detach_type !=
+      if (((ue_context_p->sgs_detach_type ==
             SGS_EXPLICIT_UE_INITIATED_IMSI_DETACH_FROM_NONEPS) ||
-           (ue_context_p->sgs_detach_type !=
+           (ue_context_p->sgs_detach_type ==
```

Comment:
Same comment as above, please flip the loop logic as you flip the equality check here.